StyleCop.Analyzers/StyleCop.Analyzers/SpacingRules/SA1005CodeFixProvider.cs

```diff
@@ -49,11 +50,17 @@ public override FixAllProvider GetFixAllProvider()
                 if (!text.StartsWith("//"))
                     continue;
 
-                string correctedText = "// " + text.Substring(2);
-                SyntaxTrivia corrected = SyntaxFactory.Comment(correctedText).WithoutFormatting();
-                Document updatedDocument = context.Document.WithSyntaxRoot(root.ReplaceTrivia(trivia, corrected));
-                context.RegisterCodeFix(CodeAction.Create("Insert space", t => Task.FromResult(updatedDocument)), diagnostic);
+                context.RegisterCodeFix(CodeAction.Create("Insert space", t => GetTransformedDocument(context, root, trivia, text)), diagnostic);
```

Comment:
:exclamation: Move as much of the logic from this loop to the callback as possible (in particular, the call to `ToFullString` above).